projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts

```diff
@@ -24,50 +24,146 @@ import { IgxTabItemTemplateDirective } from './tabs.directives';
 
 export class IgxTabsGroupComponent implements AfterContentInit, AfterViewChecked {
     private _itemStyle = 'igx-tabs-group';
+    /**
+     * @hidden
+     */
     public isSelected = false;
 
+    /**
+    * An @Input property that sets the value of the `label`.
+    *```html
+    *<igx-tabs-group label="Tab 1" icon="folder">
+    *```
+    */
     @Input()
     public label: string;
 
+    /**
+    * An @Input property that sets the value of the `label`.
+    *```html
+    *<igx-tabs-group label="Tab 1" icon="home">
+    *```
+    */
     @Input()
     public icon: string;
 
+    /**
+    * An @Input property that allows you to enable/disable the `isDisabled`.
+    *```html
+    *<igx-tabs-group label="Tab 2  Lorem ipsum dolor sit" icon="home" isDisabled="true">
+    *```
+    */
     @Input()
     public isDisabled: boolean;
 
-    @HostBinding('attr.role') public role = 'tabpanel';
-
+    /**
+     * @hidden
+     */
+    @HostBinding('attr.role')
+    public role = 'tabpanel';
+
+    /**
+     * An accessor that returns the class of the tab.
+     * ```html
+     * @ViewChild("MyTab")
+     * public tab: IgxTabsGroupComponent;
+     * ngAfterViewInIt(){
+     *    let tabStyleClass = this.tab.styleClass;
+     *    this.cdr.detectChanges();
```

Comment:
Do we need to put cdr.detectchanges() in the code snippets ? Maybe only if we are setting something.

Reply:
@hanastasov I'm not exactly sure, but for this particular snippet it wouldn't work without the cdr.detectchanges().

Reply:
There is no change to detect here, so cdr.detectchanges() is excessive.